crates/rolldown/src/ast_scanner/side_effect_detector/mod.rs

```diff
@@ -432,7 +432,7 @@ impl<'a> SideEffectDetector<'a> {
 
       Expression::ArrayExpression(expr) => self.detect_side_effect_of_array_expr(expr),
       Expression::NewExpression(expr) => {
-        let is_pure = maybe_side_effect_free_global_constructor(self.scope, expr) || expr.pure;
+        let is_pure = expr.pure || maybe_side_effect_free_global_constructor(self.scope, expr);
```

Comment:
`maybe_side_effect_free_global_constructor(self.scope, expr);` is not removed because we need to change the test setup to apply pure.